lib/manageiq/appliance_console/saml_authentication.rb

```diff
@@ -0,0 +1,212 @@
+require "uri"
+
+module ManageIQ
+  module ApplianceConsole
+    class SamlAuthentication
+      MELLON_CREATE_METADATA_COMMAND = "/usr/libexec/mod_auth_mellon/mellon_create_metadata.sh".freeze
+
+      HTTPD_CONFIG_DIRECTORY = "/etc/httpd/conf.d".freeze
+      SAML2_CONFIG_DIRECTORY = "/etc/httpd/saml2".freeze
+      IDP_METADATA_FILE      = "#{SAML2_CONFIG_DIRECTORY}/idp-metadata.xml".freeze
+
+      attr_accessor :host, :options
+
+      def initialize(options)
+        @options = options
+      end
+
+      def configure(host)
+        @host = host
+        validate_saml_idp_metadata_option
+
+        say("Configuring SAML Authentication for https://#{host} ...")
+        copy_apache_saml_configfiles
+        Dir.mkdir(SAML2_CONFIG_DIRECTORY) unless File.exist?(SAML2_CONFIG_DIRECTORY)
+        Dir.chdir(SAML2_CONFIG_DIRECTORY) do
```

Comment:
chdir is built into AwesomeSpawn (because it's built into Kernel#spawn).  Pass `:chdir => path`